### DISCUSSION TITLE
Events Page Slow to Load

### DISCUSSION THREAD

**Discussion by esmoyer** [blakeblackshear/frigate#3199] [Ask A Question] [answered]
Not sure if it's just me or anyone else; but is anyone else having issues with the events page taking a while to load? I searched the docs and issues but didn't find a whole lot, so might just be me. I'm not seeing any errors or anything like that in the logs and everything else is working fine. 

I have tried more memory and CPU but it doesn't make a huge difference. I am thinking it's DB related, because if I completely wipe all events and the DB file, it's pretty snappy for a couple weeks until it starts to get a lot of events. (My DB is on a local SSD). Is there a way I should optimize or is there a specific CPU or memory amount that would help? It's not a huge deal, but when it starts to take 20-30 seconds to load it gets a little tedious so I figured it wouldn't hurt to ask in case I was missing something.

**Comment by NickM-27** [comment, marked as answer]
This will be improved in the next release

**Reply by esmoyer** [reply]
Wow, thanks for the quick response! Is this tracked somewhere? I am curious if it's hardware or just the nature of how things are stored in the database. Thanks again!

**Reply by NickM-27** [reply]
The events page is more efficient and also there's some database optimizations that are going to be made per #2761
